File: labrabota8-gr12-MoonLory/Example2/app/src/main/java/com/example/example2/MainActivity.java
package com.example.example2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GestureDetectorCompat;

import android.os.Bundle;
import android.widget.*;
import android.view.*;

public class MainActivity extends AppCompatActivity {

    private GestureDetectorCompat mDetector;
    private TextView tvOut;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mDetector = new GestureDetectorCompat(this, new MyGestListener());
        tvOut = (TextView)findViewById(R.id.textView1);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event){
        this.mDetector.onTouchEvent(event);
        return super.onTouchEvent(event);
    }

    class MyGestListener extends GestureDetector.SimpleOnGestureListener {
        @Override
        public boolean onFling(MotionEvent event1, MotionEvent event2,
                               float velocityX, float velocityY) {
            tvOut.setText("onFling: " + event1.toString()+event2.toString());
            return true;
        }
    }

}

File: labrabota9-gr12-MoonLory/Lab9_3_3/app/src/main/java/com/example/lab9_3_2/MainActivity.kt
package com.example.lab9_3_2

import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*



class MainActivity : AppCompatActivity() {
    private lateinit var dbHandler: DBHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        addBtn.setOnClickListener { _ -> addRecord() }
        shwBtn.setOnClickListener { _ -> showRecords() }
        chgBtn.setOnClickListener { _ -> changeRecord() }

        dbHandler = DBHelper(this, null, null, 2)
    }

    private fun addRecord(){
        val mark = etCatMark.text.toString()
        val number = etCarNum.text.toString()
        val color = etCarCol.text.toString()
        val year = etCarYear.text.toString().toInt()
        val cost = etCarCost.text.toString().toInt()

        dbHandler.addNewItemToTable(Car(1, mark, number, color, year, cost))
    }

    @SuppressLint("SetTextI18n")
    private fun showRecords(){
        startActivity(Intent(this, DataActivity::class.java))
    }

    private fun changeRecord(){
        val list = dbHandler.getTable()

        val mark = etCatMark.text.toString()
        val number = etCarNum.text.toString()
        val color = etCarCol.text.toString()
        val year = etCarYear.text.toString().toInt()
        val cost = etCarCost.text.toString().toInt()


        for (i in list.indices) {
            if(list[i].number == number) {
                list[i].mark = mark
                list[i].color = color
                list[i].year = year
                list[i].cost = cost

                dbHandler.updateItem(list[i])
                break
            }
        }
    }
}